test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -6,13 +6,71 @@
 import unittest
 import sys
 import onnxruntime  # noqa
-
+import torch
+import numpy as np
+import io
 
 class TestONNXRuntime(unittest.TestCase):
 
     def test_onnxruntime_installed(self):
         self.assertTrue('onnxruntime' in sys.modules)
 
+    # There is a difference between PyTorch and ONNX/numpy w.r.t index_select by a scaler index.
+    # With an input tensor of rank r and a scaler index, PyTorch index_select returns a tensor of rank = r.
+    # However, corresponding op in ONNX(Gather) and numpy returns a tensor of rank = r - 1.
+    # To maintain a parity between ONNX and PyTorch, scaler index is converted to a 1D tensor
+    # before applying an ONNX gather op during ONNX export. 
+    # Following test_index_select_* tests are to confirm that equivalence between PyTorch and ONNX 
+    # is maintained in the above case.
+    def test_index_select_constant_scaler_index(self):
+        index = 2
+
+        class IndexSelectScalerIndexModel(torch.nn.Module):
+            def forward(self, x):
+                return torch.index_select(x, 1, torch.tensor(index))
+
+        model = IndexSelectScalerIndexModel()
+        x = torch.randn(3, 4)
+        pt_out = model(x)
+
+        f = io.BytesIO()
+        torch.onnx._export(model, (x,), f)
```

Comment:
Feel free to add some helper function (e.g. assertONNXModel(input, pytorch_model, expect_output)) like what we did in test_pytorch_onnx_caffe2.py. I think it will help you simplify the test file.

Reply:
ort tests now use a test and verify helper